### PR TITLE
Swagger UI template - use minimalist escaping for XSS prevention

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

At the bottom of https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#HTML_entity_encoding

> An alternative to escaping and unescaping JSON directly in JavaScript, is to normalize JSON server-side by converting '<' to '\u003c' before delivering it to the browser.

`{{ ...|escape('js') }}` was unnecessary in the first place because we know we're dealing with well-formed JSON, which has no possibility of causing any harm in a JS context.

### CODE REVIEW

**Comment by dunglas** [comment]
I don't feel confortable having such security related features directly in a Twig template. Does `|escape('js')` hurts?

File: src/Bridge/Symfony/Bundle/Resources/views/SwaggerUi/index.html.twig [hunk 1]
```diff
@@ -48,11 +48,11 @@
                     });
 
                     {% if operationId is not null %}
-                        {% set domId = '#' ~ shortName ~ '_' ~ operationId %}
```
Comment (dunglas) [inline src/Bridge/Symfony/Bundle/Resources/views/SwaggerUi/index.html.twig]: Why this change?

**Comment by teohhanhui** [comment]
As explained, `|escape('js')` is useless when we know we're dealing with well-formed JSON.

File: src/Bridge/Symfony/Bundle/Resources/views/SwaggerUi/index.html.twig (hunk 1 above)
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/Resources/views/SwaggerUi/index.html.twig]: Uhh... This just makes the generated code easier to read, with no change in functionality or security implications. :smile:

**Comment by dunglas** [comment]
It looks overkill to me. `escape('js')` is simple, known by most Symfony devs and AFAIK doesn't cause performance issues. Why should we change to another thing even if, indeed, it works and is enough (but isn't builtin in Twig).

**Comment by teohhanhui** [comment]
Except, `|escape('js')` does not add *even a tiny bit* of extra security in our case. What we do need is to escape for HTML. You could say this PR have at least made it a little safer. :smile:

**Comment by teohhanhui** [comment]
I'm okay if we want to take this route: https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#HTML_entity_encoding

**Comment by teohhanhui** [comment]
> Why should we change to another thing even if, indeed, it works and is enough (but isn't builtin in Twig).

We can aim for including that in Twig. I can see that `|escape('json')` is something that's missing.

**Comment by teohhanhui** [comment]
http://benalpert.com/2012/08/03/preventing-xss-json.html

@dunglas Look what I've found: https://github.com/twigphp/Twig/issues/1316#issuecomment-32325050
You made the same argument in 2014. :stuck_out_tongue:

**Comment by teohhanhui** [comment]
https://wiki.php.net/rfc/escaper

> Javascript escaping applies to all literal strings and digits. It is not possible to safely escape other Javascript markup.

Same applies for Twig.

**Comment by dunglas** [comment]
So, according to my old comment, can't we just insert the JSON as is? I'm afraid that both methods will throw a warning in SensioLabs Insight (IIRC).

**Comment by teohhanhui** [comment]
@dunglas It's safe, unless `JSON_UNESCAPED_SLASHES` is used. I'm okay with just inserting it, since we don't use that option. (But of course that might change... Unescaped slashes are perfectly valid in JSON strings.)

**Comment by dunglas** [comment]
Ok let's do this.
But can you run a SensioLabs Insight inspection before? I don't want to merge this if it adds a warning.

**Comment by teohhanhui** [comment]
@dunglas SensioLabs Insight analysis here: https://insight.sensiolabs.com/projects/fdaed34f-162f-47cd-a587-01bb2b74978c/analyses/7

**Comment by dunglas** [comment]
Thanks @teohhanhui
